ui/lib/core/addon/components/choose-pgp-key-form.js

```diff
@@ -0,0 +1,40 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+const pgpKeyFileDefault = () => ({ value: '' });
+
+/**
+ * @module ChoosePgpKeyForm
+ * ChoosePgpKeyForm component is used for DR Operation Token Generation workflow. It provides
+ * an interface for the user to upload or paste a PGP key for use
+ *
+ * @example
+ * ```js
+ * <ChoosePgpKeyForm @onCancel={{this.reset}} @onSubmit={{handleGenerateWithPgpKey}}>
+ * ```
+ * @param {function} onCancel - This function will be triggered when the modal intends to be closed
+ * @param {function} onSubmit - When the PGP key is confirmed, it will call this method with the pgpKey value as the only param
+ */
+export default class ChoosePgpKeyForm extends Component {
+  @tracked pgpKeyFile = pgpKeyFileDefault();
```

Comment:
I'm curious why this is a function instead of just the object `{value: ''}` - does it have to do with how ember tracks objects?

Reply:
Yes, you have to do the same thing with arrays IIRC. Although I'm not sure if that was a pre-Glimmer thing or if it's still required

Reply:
Yes - I know we've gotten around it with arrays by using `addObject` to update the tracked variable, but I'm not sure if there's a similar method for objects